c7n/offhours.py

```diff
@@ -277,10 +294,41 @@ def get_local_tz(self, parts):
             self.log.warning(
                 "filter:offhours unknown tz %s for %s" % (
                     tz_spec, parts))
-                    
+
             return None
         return tz
-    
+
+
+    def is_custom(self, parts):
+        if len(parts) == 3 and parts.get("on") and parts.get("off"):
+            return True
+        else:
+            return False
+
+    def get_custom_time(self, i, now, parts):
+        if type(self).__name__ in ("InstanceOnHour", "OnHour"):
```

Comment:
Makes sense. I'll put in that change shortly once I have some time away from Pycon 😄